apheleia.el

```diff
@@ -517,20 +508,64 @@ next formatter."
            (erase-buffer)
            (insert-file-contents-literally output-fname))
 
-         (if (cdr commands)
-             ;; Forward current stdout to remaining formatters, passing along
-             ;; the current callback and using the current formatters output
-             ;; as stdin.
-             (apheleia--run-formatters (cdr commands) buffer callback stdout)
-           (funcall callback stdout)))
+         (funcall callback stdout))
        :ensure
        (lambda ()
          (ignore-errors
            (when input-fname
-             (delete-file input-fname))
+             (delete-file input-fname)))
+         (ignore-errors
            (when output-fname
              (delete-file output-fname))))))))
 
+(defun apheleia--run-formatter-function (func buffer callback stdin)
+  "Run a formatter using a Lisp function FUNC.
+See `apheleia--run-formatters' for a description of BUFFER, CALLBACK
+and STDIN."
+  ;; Will be an ugly name if you use a lambda for FUNC, instead of a symbol.
+  (let ((scratch (get-buffer-create (format " *apheleia-%s-scratch*" func))))
```

Comment:
Might this have a race condition similar to what you fixed in your other PR, in case multiple formatter functions are running in parallel? We might want to implement the same fix from there (and then might as well make the buffer name prettier, e.g. by using `lambda` when the function is not a symbol).

Reply:
Ah, yep, nice spot.